luarules/gadgets/game_disable_unit_sharing.lua

```diff
@@ -17,21 +17,77 @@ if not gadgetHandler:IsSyncedCode() then
 	return false
 end
 
-if not (Spring.GetModOptions().disable_unit_sharing
-	-- tax force enables this
-	or (Spring.GetModOptions().tax_resource_sharing_amount or 0) ~= 0)
-	-- unit market handles the restriction instead if enabled so that selling still works
-	or Spring.GetModOptions().unit_market then
+local tax_resource_sharing_enabled = Spring.GetModOptions().tax_resource_sharing_amount ~= nil and Spring.GetModOptions().tax_resource_sharing_amount > 0
+local disable_share_econ_and_lab = Spring.GetModOptions().disable_unit_sharing_economy_and_production or tax_resource_sharing_enabled
+local disable_share_combat_units = Spring.GetModOptions().disable_unit_sharing_combat_units
+local disable_share_all = Spring.GetModOptions().disable_unit_sharing_all
+
+if not disable_share_econ_and_lab and not disable_share_combat_units and not disable_share_all then 
 	return false
 end
 
 
-function gadget:AllowUnitTransfer(unitID, unitDefID, fromTeamID, toTeamID, capture)
 
+-- create a table of all mex and geo unitDefIDs
+local isEconOrLab = {} 
+local isCombatUnitOrTacticalBuilding = {} 
+
+
+	-- List storages manually (some units or buildings may provide e or m storage but they are not primarily econ)
+	local storageNames = {
+		"armestor", "corestor", "legestor", "armuwes", "coruwes", "leguwes", "armuwadves", "coruwadves", "leguwadves",
+		"armmstor", "cormstor", "legmstor", "armuwms", "coruwms", "leguwms", "armuwadvms", "coruwadvms", "leguwadvms",
+	}
```

Comment:
perhaps a "dedicated storage" customparam?

Reply:
Sounds good.

Reply:
implemented it then realized unitgroup "metal" and "energy" already exist. b46f7c85007f2123fe58a5b6426a9e63f14f1e43